sprint03/ui/navbar

AfaqAhmed-XGrid opened the pull request.

AfaqAhmed-XGrid committed:
first commit

AfaqAhmed-XGrid committed:
Created top navbar, side navbar

first commit

resolved some minor issues

changed ui of signin and signup

removed eslint error

changed ui for forgot password

minor changes

completed side navbar and top navbar

removed unwanted files

solved some eslint issues

minor changes in ui

added form validation

removed unwanted files and made navbar responsive

deleted unwanted files

improved ui design

fixed some ui issue

resolved minor issues

Added new eslint rules
deleted cypress config file (not related to this PR)
deleted an empty file that was mistakenly created

resolved minor issues

junaidtariqxgrid reviewed (changes requested):
Please write a detailed description about the minor issues you have resolved

junaidtariqxgrid reviewed (changes requested):
Please write a detailed description about the UI issues you have resolved

mhassanjavedd commented on ui/src/ui/components/SideNavbar/SideNavbar.css:
```diff
@@ -126,7 +129,8 @@
   .side-nav-main-container {
-    width: 3.5rem;
+    width: 3.45rem;
+  top: 1.7rem;
```
same^

AfaqAhmed-XGrid replied:
Done

mhassanjavedd commented on ui/src/ui/components/SideNavbar/SideNavbar.css:
```diff
@@ -93,6 +95,7 @@
 @media screen and (min-width: 1440px) {
   .side-nav-main-container {
     width: 9rem;
+  top: 4.8rem;
```
Please align this property.

AfaqAhmed-XGrid replied:
Done

mhassanjavedd reviewed (commented):
Please resolve the minor alignment issues.

waqar-360 commented:
Please Merge all commits into one commit.

mhassanjavedd reviewed (approved):
looks good.

aqib-xgrid commented on ui/package.json:
```diff
@@ -3,23 +3,31 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@reduxjs/toolkit": "^1.9.5",
```
Make sure to use fixed versions of dependencies.

AfaqAhmed-XGrid replied:
Done

aqib-xgrid commented on ui/src/index.tsx:
Import comments missing

AfaqAhmed-XGrid replied:
Done

aqib-xgrid commented on ui/src/ui/components/Navbar/Navbar.css:
```diff
@@ -0,0 +1,179 @@
+.profile-displayName {
+  color: #888080;
+  font-weight: 600;
+  letter-spacing: 1px;
```
Why are you not using rem here?

AfaqAhmed-XGrid replied:
Done

aqib-xgrid reviewed (changes requested):
Kindly made these few changes.

asad-xgrid commented on ui/src/App.css:
```diff
@@ -1,38 +1,123 @@
+.form-title {
+  text-transform: uppercase;
+  font-size: 1.6rem !important;
+  letter-spacing: 2px !important;
```
use rem instead of px

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/App.css:
```diff
@@ -1,38 +1,123 @@
+}
+
+.border-test {
+  border: 1px solid black
```
same

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/App.css:
```diff
@@ -1,38 +1,123 @@
-    transform: rotate(360deg);
-  }
+.error-message {
+  color: red !important;
```
Make a separate constant file just for colors and import it from there

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/ui/components/Navbar/Navbar.css:
```diff
@@ -0,0 +1,179 @@
+.nav-title {
+  text-transform: uppercase;
+  margin: 0 !important;
+  color: #D3D3D3;
```
same

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/ui/components/Navbar/Navbar.css:
```diff
@@ -0,0 +1,179 @@
+  width: 8rem;
+  padding: 0.2rem;
+  cursor: pointer;
+  border: 2px solid #00645F;
```
Make these changes through out the app

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/ui/components/Navbar/Navbar.css:
```diff
@@ -0,0 +1,179 @@
+  }
+}
+
+@media screen and (max-width: 675px) and (min-width: 450px) {
```
Suggestion: You can also define the breakpoints(eg: phone, tablet, desktop) globally and than use them here

asad-xgrid commented on ui/src/ui/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,107 @@
+
+  const user: { email: string; displayName: string; profilePicture?: string } =
+    {
+      email: 'afaq.ahmed@xgrid.co',
+      displayName: 'Afaq Ahmed',
```
These should come from constants

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/ui/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,107 @@
+   * Function logout user
+   */
+  const onLogout = async () => {
+    toast.success('You are logged out successfully', {
+      duration: 3000,
+      position: 'bottom-center',
+      ariaProps: {
+        'role': 'status',
+        'aria-live': 'polite',
+      },
+    });
+  };
```
You can create a separate component that displays a toast on the given properties
Properties could be message and error or success toast

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/ui/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,107 @@
+
+  return (
+    <div className="nav-main-container">
+      <div style={{ backgroundColor: '#01645F' }}>
```
Avoid inline styling use class instead

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui/src/ui/components/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,107 @@
+  return (
+    <div className="nav-main-container">
+      <div style={{ backgroundColor: '#01645F' }}>
+        <img src="/assets/logo.png" alt="" className="logo" />
```
the path should come from constants

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid committed:
changes in constants, responsive, package versions

used colors as variables throughout the app. improved responsive by changing side nav bar on mobile screen, created toast function to toast messages, locked all packages versions

asad-xgrid commented on api/index.js:
```diff
@@ -16,17 +16,13 @@ const constants = require('./constants/constants');
 
 // Routes import
 const authRouter = require('./modules/auth/auth.route');
-const movieRouter = require('./modules/movie/movie.route');
```
You need to add a license in the beginning of each file
/*
 Copyright (c) 2023, Xgrid Inc, http://xgrid.co

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid committed:
added license in every file

added license in every file

also adjusted the margin top of sidenavbar in responsiveness

AfaqAhmed-XGrid committed:
Merge branch 'master' into sprint01/ui/navbar
